### PR TITLE
Fast forward on long press

### LINKED ISSUE #345: [Feature Request] Add long press gesture（添加长按手势）
# Could add a long press gesture?

Specific instructions: 

- Long press and hold the screen to play the video in multiple.
- Release your hand and stop playing in multiple.

I think this feature can be included as an optional option in the settings, which is very useful and can optimize the experience of playing videos

I hope you can accept this proposal. Thank you

---

**Example:**

https://github.com/anilbeesetti/nextplayer/assets/87907576/ae9982d1-f41c-46c3-b087-17796af872b2

### PR DESCRIPTION
**Pull request by Yashraj254**
- Fixes #345 added a feature to play the video at different speeds (default : 2x speed) while long press and hold the player screen, this feature can be enabled or disabled from the player settings screen

https://github.com/anilbeesetti/nextplayer/assets/52420957/54c53da7-828a-4d08-be93-afd2d5a8f2e4

### CODE REVIEW

**Comment by anilbeesetti** [comment]
@Yashraj254 Could you use Coroutine instead of handler

**Comment by anilbeesetti** [comment]
And the gesture also works with two fingers. Make it only activate with one finger

**Comment by Yashraj254** [comment]
@anilbeesetti i am not able to make it work for the case, when user is holding the screen with two fingers and if he lift up one of his finger while holding the screen with the other one , it won't work unless he releases his other finger as well and tap and hold the screen again...
so any suggestions to make it work for that case as well..?

File: feature/player/src/main/java/dev/anilbeesetti/nextplayer/feature/player/utils/PlayerGestureHelper.kt [hunk 1]
```diff
@@ -251,6 +254,26 @@ class PlayerGestureHelper(
         }
     )
 
+    private val longPressHoldGestureDetector = GestureDetector(
```
Comment (anilbeesetti) [inline feature/player/src/main/java/dev/anilbeesetti/nextplayer/feature/player/utils/PlayerGestureHelper.kt]: Maybe merge this into `tapGestureDetector`

Comment (Yashraj254) [reply feature/player/src/main/java/dev/anilbeesetti/nextplayer/feature/player/utils/PlayerGestureHelper.kt]: i tried it, but didn't work..

**Comment by Yashraj254** [comment]
@anilbeesetti i have made some changes to it, it works now.

**Comment by anilbeesetti** [comment]
@Yashraj254 Unfortunately, i am putting this pr on hold until the issue #459 is fixed since this pr also adds the slider dialog. some users are reporting crash on changing the slider value. I need to look into the crash issue first. Btw does slider crash issue happen on your side too or everything good

**Comment by Yashraj254** [comment]
@anilbeesetti No, no crashes, not even for once

**Comment by Yashraj254** [comment]
@anilbeesetti how about using Firebase Crashlytics in this app...

**Comment by anilbeesetti** [comment]
No way, i'm against that proprietary crap

**Comment by ShareASmile** [comment]
> how about using Firebase Crashlytics in this app..

belonging to open source community and translation contributor of this app I would oppose adding proprietary tracking libraries into the app

**Comment by Yashraj254** [comment]
> No way, i'm against that proprietary crap

@anilbeesetti ok no problem,  btw is there any specific reason behind why not to use it.. i haven't personally used it before, all i know it helps in tracking app crashes..

**Comment by anilbeesetti** [comment]
Because, it's not open source and it's google product. And doesn't align with foss principles

**Comment by Yashraj254** [comment]
Ok, got it.

**Comment by ShareASmile** [comment]
Firebase Crashlytics has foss alternatives see https://github.com/z-huang/InnerTune/issues/935#issuecomment-1666472194

**Comment by drogga** [comment]
The reason for not merging this seems now resolved (for those affected in the first place) and the issue ticket is closed, so maybe you can merge this PR, I like this feature in XPlayer and it will be nice if this awesome player has it to.

@anilbeesetti Please read your E-Mails, there should be at least ~2 to 4 with some suggestions...
Thank you all in advance.
[if this comment is problematic, I will delete it, no problem, don't worry ;)]

**Comment by anilbeesetti** [comment]
@drogga do suggestions, issues in github only. And this will be merged as soon as possible

**Comment by drogga** [comment]
It won't harm\hurt if you read them (3 E-Mails total) anyway, just to get a general idea of what they are about, but **OK, no worries, I will post them here in the discussions tab**, because I don't like creating issue tickets in GitHub, but prefer E-Mails (I have my reasons and not having PMs\DMs here doesn't help).

Thanks once again, looking forward to the version with this implemented ;)

**Comment by Yashraj254** [comment]
@anilbeesetti hey, i think it can be merged now,or do you still see any app crashes or any other issues..

**Comment by Yashraj254** [comment]
my bad, i forgot to test it when the video is already playing at a different speed other than the default(1.0f) speed..

**Comment by anilbeesetti** [comment]
@Yashraj254 Thank you for implementing this feature

**Comment by Yashraj254** [comment]
@anilbeesetti thanks to you too as you worked more than me on this feature and i got to learn more about covering the edge cases..

**Pull request merged**